### ISSUE TITLE
Formatting menu not available, and copied text not plain

### ISSUE THREAD

**Issue by fredcooke** [dylex/slack-libpurple#66]
Regularly in pidgin I'll copy some text from the chat window into the entry box, usually by select/centre click on Linux. Often this comes with formatting of various types eg links embellished in strange ways, so I use the format menu to "reset formatting" after selecting all. Just noticed that all items are greyed out and not usable, so to fix the issue I had to paste into a plain text app and copy back out again to lose the formatting. Eg:

(03:49:47) <font size="3">rob: and with a single stage paint job you will never worry about the clear coat peeling again</font>

vs

(03:49:47) rob: and with a single stage paint job you will never worry about the clear coat peeling again

Can we at least have the reset option even if the rest is not supported by the protocol? Or maybe force the content pasted to be reset/pasted plain? Not sure what control you have, but clearly some. I've not seen that option removed in other plugins. Checked, IRC doesn't have it, but pastes plain, so I guess fixing the pasting is the way to go?

**Comment by dylex** [comment]
I'm guessing this option is only enabled for protocols that use HTML, and slack does not.

**Comment by fredcooke** [comment]
Fair enough, but when I copy brightly coloured/formatted IRC text and paste
it, it comes out plain text. This plugin injects that font stuff (only, so
far) around it which is ugly to send into a channel and awkward to get rid
of. Something IRC is doing ditches the format on send, or paste, I don't
mind which, provided the <font> stuff is not sent into the channels :-)

**Comment by dylex** [comment]
The prpl does not have any direct ability to affect the interface and I don't see anything in the irc plugin about this. I don't see the behavior you describe, but I'd certainly consider a PR if you think something might help.

**Comment by fredcooke** [comment]
Roger! Appreciate your effort looking. Just did another experiment after
reading that, pasting text from slack into slack, then IRC, and then into
slack again. font stuff comes through in slack, but not IRC, so something's
different. I'll find the source and see what I can figure out! :-)

**Comment by EionRobb** [comment]
Whilst Pidgin obeys the prpl's desire regarding the formatting buttons, it doesn't clean up html that is pasted into the conversation.  If a prpl doesn't support formatting, it needs to strip tags.  (This is granular on settings too, if a prpl doesn't support font-size, but does support bold/italics/underline Pidgin will still provide font size information from the html paste buffer)

**Comment by tominventisbe** [comment]
I ran into this a couple of times and have some steps that reliably reproduce for me, which may make it easier to debug:

1. Open a chat window with someone, such as yourself (`Echo`).
2. Type `test` and send it.
3. Select the sent text `(01:23:45) YourName: test`, which contains colored text, and copy it.
4. Type `/edit ` and paste the copied contents after it, removing everything but the word "test" from the pasted text, and send the message.
5. Type `test2`, and send the message.

You and the recipient now see `<font size="3">test2</font>`, but only in the Slack interface - not in Pidgin. Every message you send will include these tags until you reset the markup by right-clicking the input field in Pidgin.
